src/main/java/calculator/controller/CalculatorController.java

```diff
@@ -0,0 +1,74 @@
+package calculator.controller;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import calculator.model.Calculator;
+import calculator.model.delimiter.CustomDelimiterProcessor;
+import calculator.model.delimiter.Delimiters;
+import calculator.model.InputFilter;
+import calculator.view.InputView;
+import calculator.view.OutputView;
+
+public class CalculatorController {
+	private final InputView inputView;
+	private final OutputView outputView;
+	private final Delimiters delimiters;
+	private final CustomDelimiterProcessor customDelimiterProcessor;
+	private final InputFilter inputFilter;
+	private final Calculator calculator;
+
+	public CalculatorController(
+		InputView inputView,
+		OutputView outputView,
+		Delimiters delimiters,
+		CustomDelimiterProcessor customDelimiterProcessor,
+		InputFilter inputFilter,
+		Calculator calculator
+	) {
+		this.inputView = inputView;
+		this.outputView = outputView;
+		this.delimiters = delimiters;
+		this.customDelimiterProcessor = customDelimiterProcessor;
+		this.inputFilter = inputFilter;
+		this.calculator = calculator;
+	}
+
+	public void run() {
+		String input = readInput();
+
+		String processedInput = processCustomDelimiter(input);
+		Set<Integer> numbers = extractNumbers(processedInput);
+		int result = sum(numbers);
+
+		printResult(result);
+	}
+
+	private String readInput() {
+		outputView.printReadCommand();
+		String input = inputView.readInput();
+		inputView.closeRead();
+
+		return input;
+	}
+
+	private String processCustomDelimiter(String input) {
+		Optional<String> customDelimiter = customDelimiterProcessor.extractCustomDelimiter(input);
+		customDelimiter.ifPresent(delimiters::addDelimiter);
```

Comment:
```suggestion
		customDelimiterProcessor.extractCustomDelimiter(input)
				.ifPresent(delimiters::addDelimiter);
```

변수 선언 없이 하나로 합쳐보면 어떨까요?

Reply:
사실 이 부분은 원래 합치려고 했었는데

```
1. 커스텀 구분자를 추출한다. Optional 타입을 명시해주면서 있을 수도, 없을 수도 있다는 것을 알린다.
2. 만약 존재한다면 구분자 목록에 추가한다.
```

이 2가지를 알리고 싶어서 찢어놓게 됐습니다! 제가 너무 리뷰어들을 무시(?)한 걸까요? ㅋㅋㅋㅋㅋㅋ 붙여놔도 좋을 것 같긴 합니다...!

Reply:
ifPresent가 상당히 옵셔널스러운 문법이라 변수 선언 없이도 충분히 의도가 전해진다고 생각해요!